server/data/mappers/workAndInterestMapper.test.ts

```diff
@@ -88,4 +91,77 @@ describe('workAndInterestMapper', () => {
       expect(actual.data.workExperience).toEqual(expectedWorkExperience)
     })
   })
+
+  describe('workInterest mapping', () => {
+    it('should map to Work And Interests given CIAG Induction with some job interests', () => {
+      // Given
+      const ciagInduction = aCiagInductionWithJobInterests()
+
+      const expectedWorkInterests: WorkInterests = {
+        hopingToWorkOnRelease: 'YES',
+        constraintsOnAbilityToWork: ['LIMITED_BY_OFFENSE'],
+        otherConstraintOnAbilityToWork: undefined,
+        jobTypes: ['CONSTRUCTION', 'OTHER'],
+        specificJobRoles: [
+          'General labourer',
+          'Being a stunt double for Tom Cruise, even though he does all his own stunts',
+        ],
+        updatedBy: 'ANOTHER_DPS_USER_GEN',
+        updatedAt: moment('2023-08-22T11:12:31.943Z').toDate(),
+      }
+
+      // When
+      const actual = toWorkAndInterests(ciagInduction)
+
+      // Then
+      expect(actual.data.workInterests).toEqual(expectedWorkInterests)
+    })
+
+    it('should map to Work And Interests given CIAG Induction with no job interests', () => {
+      // Given
+      const ciagInduction = aCiagInductionWithNoJobInterests()
+
+      const expectedWorkInterests: WorkInterests = {
+        hopingToWorkOnRelease: 'NOT_SURE',
+        constraintsOnAbilityToWork: ['CARING_RESPONSIBILITIES', 'OTHER'],
+        otherConstraintOnAbilityToWork: 'Generally a bit lazy',
+        jobTypes: undefined,
+        specificJobRoles: undefined,
+        updatedBy: 'ANOTHER_DPS_USER_GEN',
+        updatedAt: moment('2023-08-22T11:12:31.943Z').toDate(),
+      }
+
+      // When
+      const actual = toWorkAndInterests(ciagInduction)
+
+      // Then
+      expect(actual.data.workInterests).toEqual(expectedWorkInterests)
+    })
+
+    it('should map to Work And Interests given CIAG Induction with no record of any work interests', () => {
+      // Given
+      const ciagInduction = aCiagInductionWithNoRecordOfAnyWorkInterests()
+
+      const expectedWorkInterests: WorkInterests = undefined
+
+      // When
+      const actual = toWorkAndInterests(ciagInduction)
+
+      // Then
+      expect(actual.data.workInterests).toEqual(expectedWorkInterests)
+    })
+
+    it('should map to Work And Interests given CIAG Induction with no record of any previous work experience', () => {
```

Comment:
A good combination of scenarios. However you know it's going to break the moment we retrieve data from CIAG! 🤣

Reply:
💯 !! 🤣 
Yes, my scenarios are based on the cardinality of their fields, rather than any reasoned rationale from the actual data! 🤣 
Looking forward to it all going to 💩 once we integrate properly with them! 🙄